nomad/node_endpoint.go

```diff
@@ -944,7 +944,7 @@ func (n *Node) DeriveVaultToken(args *structs.DeriveVaultTokenRequest,
 	// setErr is a helper for setting the recoverable error on the reply and
 	// logging it
 	setErr := func(e error, recoverable bool) {
-		reply.Error = structs.NewRecoverableError(e, recoverable).(*structs.RecoverableError)
+		reply.Error = structs.NewRecoverableError(e, recoverable)
```

Comment:
Instead of omitting the type assertion, let's just check if error is nil and return early.

Otherwise we re-introduce a bug where `reply.Error` gets set even when the original error `e` was nil!